src/modules/TrainControllerSW.jsx

```diff
@@ -40,208 +41,274 @@ class TrainControllerSW extends React.Component {
     super(props);
     this.name = name;
 
+    // Module Communicaiton - Receiving Messages
+    window.electronAPI.subscribeTrainControllerMessage( (_event, payload) => {
+      console.log('IPC:TrainController: ', payload);
+
+      switch(payload.type) {
+        case 'commandedSpeed':
+          this.setState({commandedSpeed: commandedSpeed});
```

Comment:
do a bit of 
```javascript
this.setState({commandedSpeed: payload.commandedSpeed});
```

same idea for the rest and you're good w/ this pr i think

Reply:
done